packages/odf/components/attach-storage-storagesystem/attach-storage.tsx

```diff
@@ -0,0 +1,256 @@
+        type: AttachStorageActionType.SET_ERROR_MESSAGE,
+        payload: t('Required data is not available. Please try again later.'),
+      });
+      return;
+    }
+
+    if (!pvData.length) {
+      dispatch({
+        type: AttachStorageActionType.SET_ERROR_MESSAGE,
+        payload: t('No Persistent Volumes are available.'),
+      });
+      return;
+    }
+
+    if (!nodesData.length) {
+      dispatch({
+        type: AttachStorageActionType.SET_ERROR_MESSAGE,
+        payload: t('No nodes data is available.'),
+      });
+      return;
+    }
+
+    try {
+      dispatch({ type: AttachStorageActionType.SET_INPROGRESS, payload: true });
+      const hasFlexibleScaling = checkFlexibleScaling(storageCluster);
+      const isArbiterEnabled: boolean = checkArbiterCluster(storageCluster);
+
+      const replica = getDeviceSetReplica(
+        isArbiterEnabled,
+        hasFlexibleScaling,
+        createWizardNodeState(getCephNodes(nodesData, namespace) as NodeData[])
+      );
+
+      const pvs: K8sResourceCommon[] = getSCAvailablePVs(
+        pvData,
+        state.lsoStorageClassName
+      );
+      const availablePvsCount = pvs.length || 0;
+      const failureDomain = storageCluster?.status?.failureDomain;
+      const deviceSetCount = getDeviceSetCount(availablePvsCount, replica);
+      const filesystemName = `${storageCluster?.metadata?.name}-cephfilesystem`;
+      const payload = createPayload(
+        state,
+        storageCluster?.metadata?.name,
+        failureDomain,
+        replica,
+        filesystemName,
+        deviceSetCount
+      );
+      await sendPostRequest(payload);
+      onClose();
+    } catch (error) {
+      dispatch({
+        type: AttachStorageActionType.SET_ERROR_MESSAGE,
+        payload: error.message || t('An unexpected error has occured.'),
+      });
+    } finally {
+      dispatch({
+        type: AttachStorageActionType.SET_INPROGRESS,
+        payload: false,
+      });
+    }
+  }, [
+    t,
+    storageCluster,
+    storageClusterLoaded,
+    storageClusterLoadError,
+    nodesData,
+    nodesLoaded,
+    nodesLoadError,
+    pvData,
+    pvLoaded,
+    pvLoadError,
+    state,
+    namespace,
+    sendPostRequest,
+    dispatch,
+  ]);
+
+  return (
+    <>
+      <PageHeading title={t('Attach Storage')} breadcrumbs={breadcrumbs}>
+        <TextContent className="attachstorage__description">
+          <Text component={TextVariants.small}>
+            {t('Attach a new storage device class set to {{ resourceName }}', {
+              resourceName,
+            })}
+          </Text>
+        </TextContent>
+      </PageHeading>
+      <div className="attachstorage-form">
+        {storageClusterLoaded && !storageClusterLoadError && (
+          <LSOStorageClassDropdown
+            dispatch={dispatch}
+            storageCluster={storageCluster}
+            namespace={namespace}
+            state={state}
+          />
+        )}
+        {state.lsoStorageClassName && (
```

Comment:
why do we need this check when we have proper validations on footer ??
```suggestion
        {(
```

Reply:
Removed it, it was initially added as part of UX where we decided not to render the other components if there are no SCs selected.